sdk/state/ingest.go

```diff
@@ -28,6 +28,11 @@ func (c *Channel) IngestTx(txXDR, resultXDR, resultMetaXDR string) error {
 		return fmt.Errorf("transaction unrecognized")
 	}
 
+	// If the transaction is from an earlier sequence number, return.
+	if tx.SequenceNumber() < c.initiatorEscrowAccount().SequenceNumber {
+		return nil
+	}
```

Comment:
I think the change might need to be a little more involved than this. This code assumes the only transactions we're ingesting are transactions sourced by the initiator escrow account. However, we also ingest transactions sourced by the responder escrow account.

Reply:
We also need to still ingest transactions from elsewhere too, such as deposits.

Reply:
ahh woops 😓   I had this before and changed it last second
https://github.com/stellar/experimental-payment-channels/pull/312/commits/0479e0ae2c368617ffaa875d931098ef55104688

this ends the ingestion process early if the tx is from the initiator escrow and an earlier seq num.

the channel.State() moving backwards is currently only caused by the seq num going backwards (or an error occurring). If we ingest a transaction sourced by the responder it wouldn't affect the channel State

Reply:
it's actually weird that blocking the responder escrow account source didn't fail any tests, I'll add a test for that in a separate PR